cmake/Modules/FindOpenBLAS.cmake

```diff
@@ -7,5 +7,3 @@ ToolchainLibraryFinder(
   VERSION_FILE openblas_config.h
   VERSION_REGEX "OPENBLAS_VERSION \" ?OpenBLAS (([0-9]+\\.?)+) ?\""
 )
-
-# TODO also find LAPACK if needed
```

Comment:
I feel like we should make a check to see if LAPACK has been compiled into OpenBLAS or not (this is not guaranteed, it is optional for OpenBLAS, but would almost certainly indicate an error in the building of the docker image if it isnt there). The only way I can think of to do this is to use [try_compile](https://cmake.org/cmake/help/latest/command/try_compile.html)

Reply:
When you say use try_compile, I don't quite understand. I have read that docs page, but how would that guarantee it builds with LAPACK? Does it return false if it can't link against one of the options that you said to link against?

Reply:
You would pick a function that is only provided by LAPACK and write a small program that tries to call that function. If the program fails to compile and/or link then LAPACK is not provided by OpenBLAS.

Might even be able to use [CheckFunctionExists](https://cmake.org/cmake/help/latest/module/CheckFunctionExists.html) 
```cmake
# If you compiled OpenBLAS with LAPACK in it then it should have the
# sgetrf_single function in it.  So if we find that function in
# OpenBLAS then just use OpenBLAS's LAPACK. 
CHECK_FUNCTION_EXISTS(sgetrf_single OPENBLAS_HAS_LAPACK)
```